src/Aspire.Hosting.Azure/AzureResourceExtensions.cs

```diff
@@ -312,6 +313,26 @@ private static void WriteAzureOpenAIDeploymentToManifest(ManifestPublishingConte
         context.Writer.WriteString("parent", resource.Parent.Name);
     }
 
+    /// <summary>
+    /// Adds an Azure AI Search resource to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AzureAISearchResource}"/>.</returns>
+    public static IResourceBuilder<AzureAISearchResource> AddAzureAISearch(this IDistributedApplicationBuilder builder, string name)
```

Comment:
@mitchdenny - I know we removed the Open AI resource in https://github.com/dotnet/aspire/pull/2115. Do we want `AddAzureAISearch` in the app model?

Reply:
Do we need the AI prefix in public APIs?  We don't use that in either data plane (`Azure.Search.Documents`) or management plane (`Azure.ResourceManager.Search`).  The service has already rebranded several times from "Azure Search" to "Azure Cognitive Search" to "Azure AI Search" as well.

I'd recommend just calling this `AddAzureSearch` to mirror the C# experience more than the current branding.  Ditto for an `AzureSearchResource` class.

Reply:
Might as well update the naming everywhere(?)

Reply:
@eerhardt we removed `AddOpenAI(...)` because it isn't a resource that deployment tools can do anything with. Instead the expectation for something like that is that they would use `AddConnectionString(...)`. In this case however I think we would have `AddAzureSearch(...)` or `AddAzureAISearch(...)` because we would want an Azure-enabled deployment tool to provision the resource.

But we need to think through the end to end to make sure that whatever we provision is actually useable once the application is deployed. Deploying something that creates the Azure resource but still requires some kind of manual steps is something we would want to avoid.

Reply:
@mitchdenny I added a playground project. Could you take a look? Thanks

Reply:
One issue I ran into was the `SearchAadAuthDataPlaneAuthOptions.AadOrApiKey` property is internal, so I was unable to set _both_ RBAC and API key as auth option. This seems to be doable via the REST API so I'm unsure how to do it in .NET. I filed an issue about this [here](https://github.com/Azure/azure-sdk-for-net/issues/42051)

Reply:
~~One issue I ran into in the Provisioner was the `SearchAadAuthDataPlaneAuthOptions.AadOrApiKey` property is internal, so I was unable to set _both_ RBAC and API key as auth option. This seems to be doable via the REST API so I'm why it's internal and/or unsure how to do it in .NET. I filed an issue about this [here](https://github.com/Azure/azure-sdk-for-net/issues/42051). I set auth to only RBAC [here](https://github.com/dotnet/aspire/blob/31a26ceada3a7a2d471455802a034817661b2c96/src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureSearchProvisioner.cs#L50) as a workaround.~~

Figured it out, they are [internally setting auth option to 'both' if `AadAuthFailureMode` is set](https://github.com/Azure/azure-sdk-for-net/blob/2dc0ca8a6a16025011fca42c671f0052bab922a0/sdk/search/Azure.ResourceManager.Search/src/Generated/Models/SearchAadAuthDataPlaneAuthOptions.cs#L104).